engine/services/model_service.cc

```diff
@@ -659,17 +661,112 @@ cpp::result<bool, std::string> ModelService::StartModel(
 #undef ASSIGN_IF_PRESENT
 
     CTL_INF(json_data.toStyledString());
+    // Calculate ram/vram needed to load model
```

Comment:
Should we separate into another function?

Reply:
Will add TODO and do it in another PR. Still have the model estimation need to be merged.